### PR TITLE
prepare dockerfile for modelling test network

### PR DESCRIPTION
**Pull request by thaodt**

### CODE REVIEW

**Comment by escapedcat** [comment]
I know it's still in draft. Would be great if you could update the readme as well later on regarding how to start the container and other useful things.

File: src/bin/saitocli.rs [hunk 1]
```diff
@@ -162,6 +174,17 @@ pub async fn main() -> saito_rust::Result<()> {
...
+            App::new("read_tx")
+                .about("read content of the previous tx out & use the hash as next input tx")
+                .arg(
+                    Arg::with_name("filename")
+                        .short("f")
+                        .long("filename")
+                        .takes_value(true)
+                        .help("the previous tx output"),
+                ),
+        )
```
Comment (clayrab) [inline src/bin/saitocli.rs:187]: Can we add some docs/examples to the top of this file similar to the other commands?

File: src/bin/saitocli.rs [hunk 2]
```diff
@@ -251,18 +277,29 @@ pub async fn main() -> saito_rust::Result<()> {
...
         transaction.set_transaction_type(TransactionType::Normal);
 
-        // TODO: get inputs from the wallet and use the amount specified
+        // get inputs from the wallet and use the amount specified
         let mut input1 = Slip::new();
         input1.set_publickey(wallet.get_publickey());
-        input1.set_amount(0);
+        input1.set_amount(amount);
+        // input1.set_slip_ordinal(input_ordinal);
         input1.set_uuid([0; 32]);
```
Comment (clayrab) [inline src/bin/saitocli.rs:298]: hmmmm, does an input need an amount???

I honestly don't even know. In Bitcoin it's not needed but I guess in Saito it is. This is part of the reason I didn't want a single type for input/output.

Inputs should have a uuid(tx hash) and an ordinal.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM lukemathwalker/cargo-chef:latest-rust-1.56.1 as chef
```
Comment (clayrab) [inline Dockerfile:1]: I'd love some comments on this file. I can't really interpret what's going on here.

File: src/bin/saitocli.rs [hunk 4]
```diff
@@ -251,18 +277,29 @@ pub async fn main() -> saito_rust::Result<()> {
...
+        // let input_ordinal: u8 = matches
+        //     .value_of("input-ordinal")
+        //     .unwrap()
+        //     .parse()
+        //     .unwrap_or_else(|_error| {
+        //         println!("input_ordinal must be a float");
+        //         println!("got {}", matches.value_of("input-ordinal").unwrap());
+        //         std::process::exit(1);
+        //     });
+
```
Comment (clayrab) [inline src/bin/saitocli.rs]: Delete these?

**Review by clayrab — APPROVED** [review]
LGTM

**Pull request merged**